### PR TITLE
[MM-30353] Cypress/E2E: Automate backlogs - Plugins > Demo plgin functionality pt. 3 (4 test case)

### PR DESCRIPTION
**Pull request by jfrerich**
#### Summary
E2E testing for https://mattermost.atlassian.net/browse/MM-30353

**Waiting On PR Merge(https://github.com/mattermost/mattermost-plugin-demo/pull/130)**

**NOTE** this PR commit is based off https://github.com/mattermost/mattermost-plugin-demo/tree/MM-30351 and https://github.com/mattermost/mattermost-plugin-demo/pull/130
Many of the changes in this commit will disappear once that PR is merged to master.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-30353

### CODE REVIEW

File: e2e/cypress/integration/post_menu_spec.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        cy.apiRemovePluginById(pluginIdDemo, "");
+    });
+
+    it('MM-T3425 Post menu submenu items are selectable', () => {
+      // # post a test message
+      cy.postMessage('Hello')
+
+      // # get last postID
+      cy.getLastPostId().then((postId) => {
+
```
Comment (hanzei) [inline e2e/cypress/integration/post_menu_spec.ts:43]: Nit: Remove empty line

File: e2e/cypress/integration/demo_channel_spec.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      // # visit another channel
+      cy.visit(`/ad-1/channels/town-square`);
+
+      // # post a test message
+      cy.postMessage('hello, @demo_plugin')
+
+      // * verify return message
+      cy.getLastPostId().then((postId) => {
+        cy.get(`#post_${postId}`).should('contain', 'You must not talk about the demo plugin user')
+      })
```
Comment (hanzei) [inline e2e/cypress/integration/demo_channel_spec.ts:59]: Nit pick: Missing semicolon

File: e2e/cypress/integration/demo_channel_spec.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+      // # visit another channel
+      cy.visit(`/ad-1/channels/town-square`);
+
+      // # post a test message
+      cy.postMessage('hello, @demo_plugin')
+
+      // * verify return message
+      cy.getLastPostId().then((postId) => {
+        cy.get(`#post_${postId}`).should('contain', 'You must not talk about the demo plugin user')
```
Comment (hanzei) [inline e2e/cypress/integration/demo_channel_spec.ts:58]: `contain` or `have.text`?

**Review by hanzei — CHANGES_REQUESTED** [review]
I don't see a test case for MM-T3416. Is that intended?

**Comment by hanzei** [comment]
@jfrerich It a bit hard to spot the diff between https://github.com/mattermost/mattermost-plugin-demo/pull/130 and this PR. Could you please link it?

**Comment by saturninoabril** [comment]
Thanks Jason! I'll wait for https://github.com/mattermost/mattermost-plugin-demo/pull/130 to get merged before reviewing this PR.

**Pull request closed without merging**